### PR TITLE
Added code comments tp gameAssetContract.js

### PR DESCRIPTION
**Pull request by kbennett2000**
Added comments to gameAssetContract.js to help new developers understand the application logic

### CODE REVIEW

File: contract/src/gameAssetContract.js [hunk 1]
```diff
@@ -1,70 +1,100 @@
...
   const joinShape = harden({
     give: { Price: AmountShape },
     want: { Places: AmountShape },
     exit: M.any(),
   });
 
-  /** @param {ZCFSeat} playerSeat */
-  const joinHook = playerSeat => {
+  /**
+   * The function to be executed when a player joins the game.
```
Comment (dckc) [inline contract/src/gameAssetContract.js:56]:
```suggestion
   * Handle offers for invitations from `makeJoinInvitation` below.
```

We're trying to get away from "game". We should scrub it from the contract etc.

**Review by dckc — CHANGES_REQUESTED** [review]
Let's
 - clarify the code where useful
 - use comments for things that the code does _not_ say

Let's not repeat what the code says in prose that may or may not be precisely correct.

File: contract/src/gameAssetContract.js [hunk 2]
```diff
@@ -1,70 +1,100 @@
 /** @file Contract to mint and sell Place NFTs for a hypothetical game. */
 // @ts-check
 
-import { Far } from '@endo/far';
-import { M, getCopyBagEntries } from '@endo/patterns';
-import { AmountMath, AssetKind } from '@agoric/ertp/src/amountMath.js';
-import { AmountShape } from '@agoric/ertp/src/typeGuards.js';
+import { Far } from "@endo/far";
```
[the remark this answers is not in the document]
Comment (kbennett2000) [reply contract/src/gameAssetContract.js:4]: Understood

File: contract/src/gameAssetContract.js [hunk 3]
```diff
@@ -1,70 +1,100 @@
...
 
-const trace = makeTracer('Game', true);
+// Creating a tracer function for debugging purposes
+const trace = makeTracer("Game", true);
 
-/** @param {Amount<'copyBag'>} amt */
-const bagValueSize = amt => {
+/**
+ * Calculates the total value of a copy bag by summing up the quantities.
+ * @param {Amount<'copyBag'>} amt - The copy bag amount to calculate the value for.
```
[the remark this answers is not in the document]
Comment (kbennett2000) [reply contract/src/gameAssetContract.js:22]: okay

**Pull request closed without merging**